ts/Stock/Indicators/PivotPoints/PivotPointsIndicator.ts

```diff
@@ -0,0 +1,447 @@
+            );
+
+            xData.push(endTimestamp);
+            yData.push(PP[lastPP - 1].slice(1));
+        }
+
+        // We don't know exact position in ordinal axis
+        // So we use simple logic:
+        // Get first point in last range, calculate visible average range
+        // and multiply by period
+        this.endPoint = (slicedX as any)[0] + (
+            ((endTimestamp as any) - (slicedX as any)[0]) /
+            (slicedXLen as any)
+        ) * period;
+
+        return {
+            values: PP,
+            xData: xData,
+            yData: yData
+        } as IndicatorValuesObject<TLinkedSeries>;
+    }
+
+    public getPivotAndHLC(
+        values: Array<Array<number>>
+    ): [number, number, number, number] {
+        var high = -Infinity,
+            low = Infinity,
+            close: number = values[values.length - 1][3],
+            pivot: number;
+
+        values.forEach(function (p: Array<number>): void {
+            high = Math.max(high, p[1]);
+            low = Math.min(low, p[2]);
+        });
+        pivot = (high + low + close) / 3;
+
+        return [pivot, high, low, close];
+    }
+
+    public standardPlacement(
+        values: Array<number>
+    ): Array<(number|null)> {
+        var diff: number = values[1] - values[2],
+            avg: Array<(number|null)> = [
+                null,
+                null,
+                values[0] + diff,
+                values[0] * 2 - values[2],
+                values[0],
+                values[0] * 2 - values[1],
+                values[0] - diff,
+                null,
+                null
+            ];
+
+        return avg;
+    }
+
+    public camarillaPlacement(
+        values: Array<number>
+    ): Array<number> {
+        var diff: number = values[1] - values[2],
+            avg = [
+                values[3] + diff * 1.5,
+                values[3] + diff * 1.25,
+                values[3] + diff * 1.1666,
+                values[3] + diff * 1.0833,
+                values[0],
+                values[3] - diff * 1.0833,
+                values[3] - diff * 1.1666,
+                values[3] - diff * 1.25,
+                values[3] - diff * 1.5
+            ];
+
+        return avg;
+    }
+
+    public fibonacciPlacement(
+        values: Array<number>
+    ): Array<(number|null)> {
+        var diff: number = values[1] - values[2],
+            avg = [
+                null,
+                values[0] + diff,
+                values[0] + diff * 0.618,
+                values[0] + diff * 0.382,
+                values[0],
+                values[0] - diff * 0.382,
+                values[0] - diff * 0.618,
+                values[0] - diff,
+                null
+            ];
+
+        return avg;
+    }
+}
+
+interface PivotPointsIndicator{
+    endPoint: number;
+    nameBase: string;
+    plotEndPoint: number;
+    pointArrayMap: Array<string>;
+    pointClass: typeof PivotPointsPoint;
+    pointValKey: string;
```

Comment:
`endPoint` and `plotEndPoint` are not set in extend and should therefor be moved to the properties in the class pattern.